setup/ci/codebuild.md

```diff
@@ -0,0 +1,121 @@
+
+```
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "codebuild:StopBuild",
+        "codebuild:ListProjects",
+        "codebuild:StartBuild",
+        "codebuild:BatchGetBuilds"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+```
+This role has to trust the Spinnaker auth role so that the managing account can assume this role to call CodeBuild.
+You can edit your trust relationship to add the following policy.
+```
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": <Your Spinnaker Auth Role>
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+```
+
+## Configure Spinnaker to work with AWS CodeBuild
+
+Use the following Halyard commands to create an AWS CodeBuild account, enable the AWS CodeBuild integration, and re-deploy Spinnaker:
+```
+hal config ci codebuild account add $ACCOUNT_NAME \
+  --accountId $ACCOUNT_ID \
+  --assume-role $ASSUME_ROLE \
+  --region $REGION
+
+hal config ci codebuild enable
+
+hal deploy apply
+```
+
+## Configure an AWS CodeBuild stage
+
+To run an AWS CodeBuild build as part of a Spinnaker pipeline:
+
+1. Create a stage of type *AWS CodeBuild*.
+
+2. Configure the stage by selecting the AWS CodeBuild account to use to run the build, and selecting the project name
+from the dropdown list.
+
+3. (Optional) In the *Source Configuration* section, you might 
+  - Select the source artifact to use as build source. If not specified, the source configured in project is used.
+  - Specify the [source version](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_StartBuild.html#CodeBuild-StartBuild-request-sourceVersion)
+  of the build source.If not specified, the latest version is used.
+  - Specify the buildspec file. If left blank, the buildspec configured in project is used.
+  - Select secondary build sources. If not specified, the secondary sources configured in project is used
+
+4. (Optional) In the *Environment Configuration* section, you might specify the image tag or image digest that identifies the Docker image
+to use. If not specified, the image configured in project is used. **Note:** As prerequisite, follow this
+[user guide](https://docs.aws.amazon.com/codebuild/latest/userguide/sample-private-registry.html) to set up your registry credentials
+in project configuration.
+
+5. (Optional) In the *Advanced Configuration* section, you might specify environment variables to use in the build.
+Only environment variables in plain text are supported.
+
+6. (Optional) In the *Produces Artifacts* section, you might supply any artifacts that you expect the build to create in order to
+make these artifacts available to downstream stages. AWS CodeBuild supports creating S3 artifacts, which will be converted
+to Spinnaker artifacts and injected into the pipeline on completion of the build.
```

Comment:
Are there any restrictions on the artifact packaging value in the project for this, like must be set to "zip"?  Can Spinnaker artifacts handle getting an S3 folder returned if "none" packaging is specified?

Reply:
Yes I think Spinnaker could only handle s3 object but not s3 folder
https://www.spinnaker.io/reference/artifacts/types/s3-object/